src/library/dlgtrackinfo.cpp

```diff
@@ -230,10 +232,12 @@ void DlgTrackInfo::init() {
                 this,
                 &DlgTrackInfo::slotCoverFound);
     }
+
     connect(m_pWCoverArtLabel.get(),
```

Comment:
here you can directly connect the menu and remove the connection and signals from the label.
If this is the only dependency the WCoverArtLabel can take the generic menu.

Reply:
I actually tried it before, as far as I remember that was causing a bug. I click on the cover art layout, full size populated, after right click - clear cover was just clearing the cover art for the DlgTrackInfo, not in all of them which was not like in the original use-case.

Reply:
Can you try this again? I see no reason that it can cause a bug, because it is only a signal signal connection. without it we remove a redirection loop, that's all. (make sure to change the pointer AND the base class of the connected signal function pointer)

Reply:
I deleted the signal signal connection, and it works just fine, but whenever I want to pass the QMenu, i can not because they have dependency.

WCoverArtLabel takes WCoverArtMenu and DlgCoverArtFullSize, DlgCoverArtFullSize also take WCoverArtMenu, so they have highly related dependency.